### PR TITLE
Revise all e2e tests to use the kind cluster with local registry

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

The pr is to use the `local registry` of kind for `repartition dask e2e test`. 

If the repartition e2e test passed, we can consider replacing all e2e tests here as it can resolve problems like `No space left on device error on CI`. Also, the pr needs to wait for the [e2e tool](https://github.com/apache/skywalking-infra-e2e) to support for testing on a existing kubernetes cluster.

### CODE REVIEW

**Comment by sighingnow** [comment]
The dask CI looks work!

Could you please revise all k8s test cases to use the local registry?

Thanks!

**Comment by dashanji** [comment]
No problem. I have added the feature `testing on a existing kubernetes cluster` to the [e2e tool](https://github.com/apache/skywalking-infra-e2e) yesterday. Until it's merged, we can revise all e2e tests more smoothly.

File: k8s/pkg/templates/operation/dask-repartition.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ spec:
     spec:
       containers:
       - name: dask-repartition-operation
-        image: localhost:5001/dask-repartition
+        image: ghcr.io/v6d-io/v6d/dask-repartition
```
Comment (sighingnow) [inline k8s/pkg/templates/operation/dask-repartition.yaml:28]: Why ghcr instead of local registry?

Comment (dashanji) [reply k8s/pkg/templates/operation/dask-repartition.yaml]: The local registry is only for testing, for others using vineyard operator in their environment, the image should be able to be downloaded normally.

**Comment by dashanji** [comment]
The bugs in the failover demo are fixed here.

File: .github/workflows/vineyard-operator.yaml [hunk 2]
```diff
@@ -122,7 +122,13 @@ jobs:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-
+      
+      - name: Build a kind cluster with local registry
+        run: |
+          touch /tmp/e2e-k8s.config
+          export KUBECONFIG=/tmp/e2e-k8s.config
+          bash k8s/test/hack/build-kind-cluster-with-local-registry.sh
```
Comment (sighingnow) [inline .github/workflows/vineyard-operator.yaml:130]: I suggest moving the "install cert-manager" and "deploying vineyardd cluster" steps to here as well to further reduce the duplications in "xxxx/e2e.yaml"s.

**Review by sighingnow — APPROVED** [review]

File: k8s/test/e2e/repartition/dask-repartition-e2e.yaml [hunk 3]
```diff
@@ -73,11 +77,14 @@ setup:
         sed 's/$job/daskRepartitionJob2/' k8s/test/e2e/repartition-demo/repartition-job.yaml | \
           gomplate -d config=k8s/test/e2e/repartition-demo/repartition-job-config.yaml | \
           kubectl apply -f -
+        #sleep 120
+        #kubectl describe pod $(kubectl get po -l app=dask-repartition-job2 -n vineyard-job -oname | awk -F '/' '{print $2}') -n vineyard-job
+        #kubectl logs -f $(kubectl get po -l app=dask-repartition-job2 -n vineyard-job -oname | awk -F '/' '{print $2}') -n vineyard-job
```
Comment (sighingnow) [inline k8s/test/e2e/repartition/dask-repartition-e2e.yaml:82]: Remove the temporary debug lines.

File: .github/workflows/vineyard-operator.yaml (hunk 2 above)
Comment (dashanji) [reply .github/workflows/vineyard-operator.yaml:130]: Make sense

File: k8s/test/e2e/assembly/distributed-assembly-e2e.yaml [hunk 4]
```diff
@@ -25,23 +25,19 @@ setup:
         - namespace: cert-manager
           resource: pod
           for: condition=Ready
-    - name: install scheduler-plugin and operator
+    - name: install vineyard operator
       command: |
-        kind load docker-image vineyardcloudnative/vineyard-operator:latest
+        docker tag vineyardcloudnative/vineyard-operator:latest localhost:5001/vineyard-operator:latest
+        docker push localhost:5001/vineyard-operator:latest
+        export IMG=localhost:5001/vineyard-operator:latest
```
Comment (sighingnow) [inline k8s/test/e2e/assembly/distributed-assembly-e2e.yaml:32]: Why not `make install-vineyard-operator` as you have already added the target in Makefile?

**Review by sighingnow — APPROVED** [review]

**Comment by dashanji** [comment]
Done, please take another took at this. @sighingnow

**Pull request merged**